sources/packages/backend/apps/api/src/services/application-offering-change-request/application-offering-change-request.service.ts

```diff
@@ -515,20 +523,100 @@ export class ApplicationOfferingChangeRequestService {
    * @param applicationOfferingChangeRequestId application offering change request id for which to update the status.
    * @param applicationOfferingChangeRequestStatus the application offering change request status to be updated.
    * @param studentConsent student consent to approve the application offering change request.
+   * @param auditUserId user that should be considered the one that is causing the changes.
    */
   async updateApplicationOfferingChangeRequest(
     applicationOfferingChangeRequestId: number,
     applicationOfferingChangeRequestStatus: ApplicationOfferingChangeRequestStatus,
     studentConsent: boolean,
+    auditUserId: number,
   ): Promise<void> {
+    const auditUser = { id: auditUserId } as User;
+    const currentDate = new Date();
     await this.applicationOfferingChangeRequestRepo.update(
       {
         id: applicationOfferingChangeRequestId,
       },
       {
         applicationOfferingChangeRequestStatus,
         studentConsent,
+        studentActionDate: currentDate,
+        modifier: auditUser,
+        updatedAt: currentDate,
       },
     );
   }
+
+  /**
+   * Assess the application offering change request status for the given application offering change request id for the ministry user.
+   * @param applicationOfferingChangeRequestId application offering change request id for which to update the status.
+   * @param applicationOfferingChangeRequestStatus the application offering change request status to be updated.
+   * @param assessmentNote note added while updating the application offering change request.
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   */
+  async assessApplicationOfferingChangeRequest(
+    applicationOfferingChangeRequestId: number,
+    applicationOfferingChangeRequestStatus:
+      | ApplicationOfferingChangeRequestStatus.Approved
+      | ApplicationOfferingChangeRequestStatus.DeclinedBySABC,
+    assessmentNote: string,
+    auditUserId: number,
+  ): Promise<void> {
+    const auditUser = { id: auditUserId } as User;
+    const currentDate = new Date();
+    const applicationOfferingChangeRequest = await this.getById(
+      applicationOfferingChangeRequestId,
+    );
+    const application = applicationOfferingChangeRequest.application;
+    await this.dataSource.transaction(async (transactionalEntityManager) => {
+      // Save the note.
+      const noteEntity = await this.noteSharedService.createStudentNote(
+        applicationOfferingChangeRequest.application.student.id,
+        NoteType.Application,
+        assessmentNote,
+        auditUserId,
+        transactionalEntityManager,
+      );
+      // Update the application offering change request.
+      await transactionalEntityManager
```

Comment:
Since the `applicationOfferingChangeRequest` is available it can be used to group all changes and do all the DB changes with a single `save` operation.
Just ensure that the `ApplicationOfferingChangeRequest.application` has the `cascade: ["update"]` set.